src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -66,38 +70,41 @@ bool HelpPromptManagerImpl::AddCommand(
     return false;
   }
 
+  const smart_objects::SmartObject& commands = command[strings::vr_commands];
+  if (commands.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands array is empty");
+    return false;
+  }
+
   sync_primitives::AutoLock lock(vr_commands_lock_);
   auto it = std::find_if(
       vr_commands_.begin(),
       vr_commands_.end(),
       [cmd_id](const VRCommandPair& pair) { return pair.first == cmd_id; });
 
   if (vr_commands_.end() != it) {
-    LOG4CXX_DEBUG(logger_, "Commands with id:" << cmd_id << " already exists");
+    LOG4CXX_DEBUG(logger_, "Command with id:" << cmd_id << " already exists");
     return false;
   }
 
-  const smart_objects::SmartObject& commands = command[strings::vr_commands];
   const size_t count_new_commands = commands.length();
```

Comment:
`count_new_commands` should be removed at this point, it is no longer useful, since only one synonym is used for each command.

Reply:
@jacobkeeler agree. Removed in https://github.com/smartdevicelink/sdl_core/pull/2557/commits/445e711459cce157e63e355f3e20df300aa7a963